### DISCUSSION TITLE
a nice visualizer - projectm

### DISCUSSION THREAD

**Discussion by leosin** [losnoco/Cog#272] [Ideas]
I found a really good visualizer, can I incorporate it

https://github.com/projectM-visualizer/projectm

> projectM is an open-source project that reimplements the esteemed [Winamp Milkdrop](https://en.wikipedia.org/wiki/MilkDrop) by Geiss in a more modern, cross-platform reusable library.

**Comment by kode54** [comment]
You're welcome to do so, it would be a welcome addition, as long as it works with the current system, and hopefully works on more systems than the SceneKit visualizer works on. Even if it works on the same system set, that's great.

I *think* they have a Metal implementation now, too. Maybe.

Edit: Also feel free to implement a Visualization system API that only returns PCM, if Project M only needs PCM and does its own FFT/DFT.
